st2reactor/st2reactor/cmd/sensormanager.py

```diff
@@ -45,14 +45,14 @@ def main():
     try:
         _setup()
         container_manager = SensorContainerManager()
-        sensors = _get_all_sensors()
+        sensors = None
 
-        if cfg.CONF.sensor_name:
-            # Only run a single sensor
-            sensors = [sensor for sensor in sensors if
-                       sensor.name == cfg.CONF.sensor_name]
+        if cfg.CONF.sensor_ref:
+            sensors = [SensorType.get_by_ref(cfg.CONF.sensor_ref)]
             if not sensors:
```

Comment:
If `SensorType.get_by_ref` returns `None` and doesn't throw on in-existent DB object, we will never get into the if clause bellow since `bool([None]) is True`.

In short, we need to do something like:

```python
sensor = SensorType.get_by_ref(cfg.CONF.sensor_ref)
if not sensor:
    ...

sensors = [sensor]
```